scripts/build_plugin.py

```diff
@@ -322,24 +353,40 @@ def _bake_manifest(manifest_data, config_uri, core_descriptor, plugin_root):
             fh.write("BUILD_GENERATION=%d\n" % BUILD_GENERATION)
 
             # Write out helper function 'get_sgtk_pythonpath()'.
-            core_path_parts = os.path.normpath(core_descriptor.get_path()).split(os.path.sep)
-            core_path_relative_parts = core_path_parts[core_path_parts.index(BUNDLE_CACHE_ROOT_FOLDER_NAME):]
-            core_path_relative_parts.append("python")
 
-            fh.write("\n\n")
-            fh.write("def get_sgtk_pythonpath(plugin_root):\n")
-            fh.write("    import os\n")
-            fh.write("    return os.path.join(plugin_root, %s)\n" %
-                     ", ".join('"%s"' % dir for dir in core_path_relative_parts))
-            fh.write("\n\n")
+            if core_descriptor.get_path().startswith(plugin_root):
+                # the core descriptor is cached inside our plugin
+                core_path_parts = os.path.normpath(core_descriptor.get_path()).split(os.path.sep)
+                core_path_relative_parts = core_path_parts[core_path_parts.index(BUNDLE_CACHE_ROOT_FOLDER_NAME):]
+                core_path_relative_parts.append("python")
+
+                fh.write("\n\n")
+                fh.write("def get_sgtk_pythonpath(plugin_root):\n")
+                fh.write("    import os\n")
+                fh.write("    return os.path.join(plugin_root, %s)\n" %
+                         ", ".join('"%s"' % dir for dir in core_path_relative_parts))
+                fh.write("\n\n")
+
+            else:
+                # the core descriptor is outside of bundle cache
+                logger.warning("Your core %r has its payload outside the plugin bundle cache. "
+                               "This plugin cannot be distributed to others." % core_descriptor)
+
+                core_path_parts = os.path.normpath(core_descriptor.get_path()).split(os.path.sep)
+                core_path_parts.append("python")
+
+                fh.write("\n\n")
+                fh.write("def get_sgtk_pythonpath(plugin_root):\n")
+                fh.write("    return '%s'\n" % os.path.sep.join(core_path_parts))
```

Comment:
maybe add a comment in here about why the `plugin_root` isn't used just to reenforce for developers? I guess echoing the warning above?